### ISSUE TITLE
(v3.1) Automatic batching doesn't work on Android

### ISSUE THREAD

**Issue by hakonber** [cocos2d/cocos2d-objc#831]
This is using Cocos2D v3.1 final release, Apportable 1.1.10 and a Nexus 5. See screenshots below.

I've created a simple test project that puts 100 copies of the same sprite on screen. Results below:

iPhone (2 draw calls):
![ios simulator screen shot 13 jun 2014 12 12 33](https://cloud.githubusercontent.com/assets/873092/3268900/92b9f35e-f2e3-11e3-8711-2f83dd869d6a.png)

Android (112 draw calls):
![image](https://cloud.githubusercontent.com/assets/873092/3268916/e64f4c3a-f2e3-11e3-96d4-a087ea67badc.png)

To reproduce yourself, create a new Cocos2D project using the Xcode template, and replace the contents of IntroScene.m with the following:

``` objc
#import "IntroScene.h"

@implementation IntroScene

+ (IntroScene *)scene
{
    return [[self alloc] init];
}

- (id)init
{
    self = [super init];
    if (!self) return(nil);

    for (int x = 0; x < 10; x++) {
        for (int y = 0; y < 10; y++) {
            CCSprite *sprite = [CCSprite spriteWithImageNamed:@"Icon-72.png"];
            sprite.position  = ccp(40 + 30 * x, 40 + 30 * y);
            [self addChild:sprite];
        }
    }

    return self;
}

@end
```

**Comment by hakonber** [comment]
I'm pretty sure it's got something to do with the caching of CCRenderStates. I've narrowed it down to the following:

If I change the implementation of [CCRenderState copyWithZone:] like this, the batching starts working again.

``` objc
(id)copyWithZone:(NSZone *)zone
{
//  if([_shaderUniforms isKindOfClass:[NSMutableDictionary class]]){
//      return [[CCRenderState allocWithZone:zone] initWithBlendMode:_blendMode shader:_shader shaderUniforms:[_shaderUniforms copy]];
//  } else {
        return self;
//  }
}
```

This could have bad side effects for all I know...

**Comment by slembcke** [comment]
Hmm. Dictionary literals in apportable must not create NSMutableDictionary objects. (Class cluster maybe?) I guess I never noticed since it otherwise rendered correctly. O_o

**Comment by agro1986** [comment]
I'm using the latest v3.1 git branch (dc3ad27) with the latest Apportable 1.1.11 and I can confirm the bug still exists.

**Comment by slembcke** [comment]
This was fixed in 3.2.
